fontforge/encoding.c

```diff
@@ -548,38 +548,103 @@ static Encoding *ParseConsortiumEncodingFile(FILE *file) {
     memset(encs, 0, sizeof(encs));
     max = -1;
 
-    while ( fgets(buffer,sizeof(buffer),file)!=NULL ) {
-	if ( ishexdigit(buffer[0]) ) {
-	    if ( sscanf(buffer, "%x %x", (unsigned *) &enc, (unsigned *) &unienc)==2 &&
-		    enc<0x10000 && enc>=0 ) {
-		encs[enc] = unienc;
-		if ( enc>max ) max = enc;
-	    }
-	}
-    }
-
-    if ( max==-1 )
-return( NULL );
+    while (fgets(buffer, sizeof(buffer), file) != NULL) {
+        if (ishexdigit(buffer[0])) {
+            if (sscanf(buffer, "%x %x", (unsigned *) &enc, (unsigned *) &unienc)==2
+                && enc < 0x10000
+                && enc >= 0) {
+                encs[enc] = unienc;
+                if ( enc>max ) max = enc;
+            }
+        }
+    }
+
+    if (max==-1)
+        return NULL;
 
     ++max;
-    if ( max<256 ) max = 256;
-    item = calloc(1,sizeof(Encoding));
+    if (max<256) max = 256;
+    item = calloc(1, sizeof(Encoding));
     item->only_1byte = item->has_1byte = true;
     item->char_cnt = max;
     item->unicode = malloc(max*sizeof(int32));
-    memcpy(item->unicode,encs,max*sizeof(int32));
-return( item );
+    memcpy(item->unicode, encs, max*sizeof(int32));
+    return item;
+}
+
+/* Parse the GlyphOrderAndAliasDB file format */
+static Encoding *ParseGlyphOrderAndAliasDB(FILE *file) {
+    char* names[1024];
+    char buffer[256];
+    int32 encs[1024];
+    Encoding* item = NULL;
+    size_t i, any;
+    int max, enc;
+
+    for (i=0; i < sizeof(names)/sizeof(names[0]); ++i) {
+        encs[i] = -1;
+        names[i] = NULL;
+    }
+
+    max = -1; any = 0;
+    i = 0;
+    while (fgets(buffer, sizeof(buffer), file) != NULL) {
+        max = i;
+
+        int tab, tab2;
+        for (tab = 0; tab < sizeof(buffer) && buffer[tab] != '\t'; tab++);
+        if (tab < sizeof(buffer))
+            buffer[tab]='\0';
+
+        for (tab2 = tab+1; tab2 < sizeof(buffer) && buffer[tab2] != '\t'; tab2++);
+        if (tab2 < sizeof(buffer))
+            buffer[tab2]='\0';
+
+        if (strcmp(buffer, ".notdef") == 0) {
+            encs[i] = -1;
+        } else if ((enc = UniFromName(buffer, ui_none, &custom)) != -1) {
+            encs[i] = enc;
+
+            /* Used not to do this, but there are several legal names */
+            /*  for some slots and people get unhappy (rightly) if we */
+            /*  use the wrong one */
```

Comment:
We were not doing any of this before, so I'm not sure what this comment references. Are you referring to trying to look up the Unicode name?